vm_backtrace.c

```diff
@@ -1613,7 +1613,17 @@ rb_profile_frames(int start, int limit, VALUE *buff, int *lines)
                 buff[i] = (VALUE)cfp->iseq;
             }
 
-            if (lines) lines[i] = calc_lineno(cfp->iseq, cfp->pc);
+            if (lines) {
+                // If this frame has a jit_return pointer on it, then the PC
+                // _is not_ reliable.  We can get a crash if we try to calculate
+                // the line number.
+                if (cfp->jit_return) {
+                    lines[i] = 0;
```

Comment:
Could we maybe use `cfp->iseq->body->iseq_encoded` instead of `cfp->pc`?

Reply:
I don't understand.  You mean just to decode the line, rather than return 0?

Reply:
I meant decoding the first line of the iseq, assuming it's not 0? could be just `rb_iseq_first_lineno(cfp->iseq)`.

Reply:
I think we could, but it seems like `0` is more honest.  Since line numbers should start at 1, I think that 0 means "I don't know".  If we return a real line number, then it might be more confusing.

Reply:
ok, I just didn't know it starts at 1 (I thought it starts at 100 if it's 100 in the file). It seems fine then.